### DISCUSSION TITLE
Frequent pair requests on Samsung A22 with Dexcom G6

### DISCUSSION THREAD

**Discussion by scimone** [NightscoutFoundation/xDrip#1951] [Q&A]
Edit: Samsung A22 not A21, sorry!
I have been struggling with regular unpairings on my Samsung for weeks now, despite using all recommended settings.

I have tried all the troubleshooting that is described here:
https://github.com/Navid200/xDrip/wiki/Intermittent-Connectivity?fbclid=IwAR2rPZUqdzNnmuancEiuzMtU7aTO_K10_0fffBlR378qlTQMuU3EU34V4xI
Even if I disable "Allow OB1 unbonding", it disconnects.

I also tried turning off all other bluetooth devices that might disturb the connection, including my dana pump.
However, my phone still unpairs around 30 times a day.

On Monday I set up xdrip on an old Samsung S5 Mini (Android 6). It had a stable connection for 3 days in a row.
Now I switched back to A21 again and it directly disconnected after about 20 minutes ("Allow OB1 unbonding" is disabled).

Do you have any ideas what else I can try?

<img width="350" src="https://user-images.githubusercontent.com/56034784/150403767-a6b9fc90-a0a2-4377-b029-87965e213bad.jpg" /> <img width="350" src="https://user-images.githubusercontent.com/56034784/150404372-733367d4-8361-423a-9de2-e942d5a8c4a4.jpg" /> <img width="350" src="https://user-images.githubusercontent.com/56034784/150404375-9563b98e-bc10-4cc3-9a12-6a8762d41813.jpg" />

**Comment by emp-00** [comment]
@scimone : here's the solution for your problem, I just had exactly the same problem.

https://github.com/Navid200/xDrip/discussions/171

**Reply by scimone** [reply]
Thank you, I have downloaded the app now. Will test and report back tomorrow :)

**Reply by Navid200** [reply]
@Foxy7, thanks for suggesting this app.
@emp-00, thanks for bringing it up.
@scimone, please let us know if it makes a difference or not.

**Reply by scimone** [reply]
I just had another pair request. I still give it a chance, maybe it will get better.

**Reply by emp-00** [reply]
For me this works since already 2 weeks with not even a single dropout. Thanks to @Foxy7. And as already suggested, maybe this helps identifying the root cause and implementing a "workaround/bugfix" in xDrip+ ?

**Reply by scimone** [reply]
@emp-00 What are your "Allow OB1 unbonding" and "Avoid/minimize scanning" settings?

**Reply by emp-00** [reply]
@scimone : Check under Android -> Settings/Developer Options/Running services/Running services -> is the App "Blue car Tethering" listed there? This is the App preventing the BT pairing dropouts in my case. Have you installed exactly this App? Then it must show in the "Running services" list in order to work.

**Reply by scimone** [reply]
@emp-00 Yes it's listed. I have restarted my phone after the disconnect, maybe that helped

**Reply by scimone** [reply]
Another pair request

**Comment by emp-00** [comment]
@scimone - I have "only" had loss of pairing every ~24 to 48 hours but (luckily) not 40-50 times (???!) a day like you. Just saw that - never heard of such mega-serious issues before ...

Allow OB1 unbonding and Avoid/minimize scanning are all switched ON on my Samsung phone.

Before installing the App I also experimented with these settings as suggested by @Navid200, but they made no difference in my case. Only the App helped ...

**Reply by scimone** [reply]
Just noticed that my estimation was bad, it's more around 30 times per day, but still too much for my taste :D

**Reply by emp-00** [reply]
Let us know if restarting helped you fixing the issue. Keeping my fingers crossed.

**Reply by scimone** [reply]
Sadly no, it just disconnected again :/

**Reply by emp-00** [reply]
Have you (exactly!) followed this guide already? https://github.com/Navid200/xDrip/wiki/Intermittent-Connectivity
This is also mandatory and only with these steps/settings in combination with the "Blue car Tethering" app it worked on my Samsung phone.

**Reply by scimone** [reply]
Everything was the same as recommended settings except for Allow OB1 unbonding and Avoid/minimize scanning. I have switched them on now, let's see

**Reply by scimone** [reply]
Nope disconnected again

**Reply by emp-00** [reply]
100% sure your "Battery optimization" settings in Android (no optimization active for xDrip+!!) are as per the FAQ?

**Reply by scimone** [reply]
Yes I just checked

**Reply by scimone** [reply]
I saw in the reviews of blue car tethering app that it doesnt work on android 11 anymore. Which android do you have?

**Reply by emp-00** [reply]
I have Android 11 as well! And the disconnect issues started on my Samsung after receiving the A11 update. But for me the mentioned app works - even though the app is warning "may not work on never Android releases" when I first started the app. But I just clicked OK and left it running -> problem solved ever since.

**Reply by scimone** [reply]
Yes I got that warning too...
Tomorrow I will try my old Samsung S9. It also had connectivity problems since November (I guess also due to the update), but maybe the blue car app works better there.

**Reply by Foxy7** [reply]
@scimone  after installing blue car tethering, did you open the app and tick the box to force it to connect to the dexcom transmitter? (it's not clear if you've done that step from the above chat).

**Reply by scimone** [reply]
@Foxy7 Yes I did :D

**Reply by Foxy7** [reply]
i have Android 11 on one of my phones and seems to work ok. don't get the repeated requests you have. S20.

**Comment by mk886714** [comment]
since A11 I had similar problems at first, which I no longer have since I excluded xdrip, blooototh and AAPS from data saving

**Reply by mk886714** [reply]
ALLOW APPS / SPECIAL ACCESS / DATA USE WITH DATA SAVER .  Show system applications in the top right menu, activate APPS /blooototh midi service/xdrip there.  Every Samsung device has been running since I set this

**Reply by mk886714** [reply]
I also check this after every update, since it is deactivated again after most updates

**Reply by scimone** [reply]
Thank you for the hint. Xdrip was already activated, now I additionally activated the other two

**Reply by scimone** [reply]
@mk886714 @emp-00 @Navid200 @Foxy7 
Update: 
After having allowed data use with data saver for 
- Bluetooth midi service, 
- AAPS,
- Blue car tethering app,
(xdrip was already activated), 

the connection has been stable for 4h straight, which is already much longer than usual. Will keep you updated. If it's stable until tomorrow, I will also try uninstalling blue car tethering to see if it also works without. Thank you again @mk886714 for giving me the hint!

**Reply by Navid200** [reply]
So, you do use AAPS then?

**Reply by scimone** [reply]
@Navid200 Not at the moment because my pump bluetooth is broken, that's another story. But since mk886714 also enabled it, I just wanted to have the same settings.

**Reply by jamorham** [reply]
@scimone which pump type is it?

**Reply by scimone** [reply]
@jamorham It's a Dana RS. The bluetooth seems to be broken because it is not recognized by any phone

**Reply by mk886714** [reply]
@scimone 
Samsung operating systems have always reduced data consumption in order to save electricity and to reserve the performance for display and sound, in order to try to display the device better than other manufacturers with the same computing power

**Reply by mk886714** [reply]
I would also advise against using an additional app (blue car tethering) as this only affects your Bloodoth settings.  So another additional source of interference for your already unstable connection.
!  my opinion !

**Reply by mk886714** [reply]
If you don't use AAPS you can also omit this.
Maybee your dana is still connected to another device?

**Reply by scimone** [reply]
> I would also advise against using an additional app (blue car tethering) as this only affects your Bloodoth settings. So another additional source of interference for your already unstable connection. ! my opinion !

@mk886714 Yeah as I said, I first want to confirm that my current setup is stable for 24h, then I can try to tweak stuff again by uninstalling blue car tethering. 
By the way, still not a single disconnect so far!

**Reply by scimone** [reply]
> If you don't use AAPS you can also omit this. Maybee your dana is still connected to another device?

No it's not connected to any device

**Reply by emp-00** [reply]
@mk886714 @scimone / All ==> In my case I have Settings / "Data saver" => OFF. That's probably why I don't need to do the **"Settings Special Access / Allowed to use data while Data saver" activated for "Bluetooth Midi"** ? The latter I just activated as well - cannot hurt, I guess!

@mk886714 : Thanks for your hints! Maybe with your above advice using the "Blue Car Tethering App" is not needed any more? However, I don't think so - because I have "Data saver" completely OFF, see above. But who knows?!
@Navid200 : Would you care to add these important tricks in your FAQ for Samsung/G6?

**Reply by scimone** [reply]
@emp-00 Interestingly, my data saver is also off...

**Reply by Navid200** [reply]
As soon as there is confirmation that any of them works, absolutely.

At this point, it seems like a lot of different suggestions with no way to confirm which one is really necessary and which one is just a coincidence.

**Reply by scimone** [reply]
@Navid200 Is there any specific combination of settings you want me to test next?

**Reply by scimone** [reply]
BTW I disabled "minimize/avoid scanning" and "allow ob1 unbonding" again yesterday. Because when they were on for some time, my phone disconnected without giving me new pair requests. Should I change some of these settings again, and if yes, which ones?

**Reply by Navid200** [reply]
@scimone No, that message was in response to emp-00.
All I hope is that you get your system up and running in a satisfactory way.

**Reply by emp-00** [reply]
My recommendation: Let's try to use Navid200's standard settings as per his FAQ as starting point. This is what works for "all other systems". For our Samsung's I would focus on the above mentioned App/Special/Data+Battery-Saving-Options: Let's try to find out which combination of these Samsung-Settings does the trick. Then Navid200 could add a "Samsung/G6-Settings"-section in his FAQ which is hopefully compatible with the other xDrip-settings.

I have now switched all settings as per @mk886714 suggestions (one additional check for "Bluetooth midi"), If everyting remains OK in the next days I will remove the "Blue Car Tethering" app in order to see if this is maybe not needed any more.

**Reply by mk886714** [reply]
At all  ...... I can't say why, but it was only with this setup that my connection failures disappeared
I've been using ruffy from the start (Accu check Combo), it has a very fragile bloototh connection (BLE), so I had to test all the setup options in the Samsung device.  Turning on NFC and location also improves reconnecting in my case, but this takes a lot of battery power (as a next option)
I suspect this is due to the chipsets installed by Samsung in connection with their own operating system, since lineageOS also has similar problems with the Samsung peripherals

**Reply by scimone** [reply]
Update: First pair request after 14 hours of silence.

**Reply by scimone** [reply]
Another one. It might have to do with my battery which is <10%

**Reply by mk886714** [reply]
By the was to all, if your samsung phone is older than 20 months then reset the loading counter.  You can google the code.  I didn't believe this at first either, but the charge on my cell phones lasts a good 4-6 hours longer

**Reply by scimone** [reply]
@mk886714 No it's only a month old

**Comment by scimone** [comment]
So after 14 hours of straight connection, I got pair requests again at the same frequency as before. They began exactly at midnight when the battery was low, but remained even after charging. Honestly I have no more ideas what to try.

**Reply by scimone** [reply]
@Navid200 
Two more things I've noticed:
- The pair requests only occur while I am not using my phone (= when it's on standby).
- They seem to only occur when I'm at home. Not 100% sure but I can't remember any while I was outside. Today for example I was outside of home for 5h and in that time, not a single pair request. As soon as I entered the door, I got a pair request again. (I spent a lot time at home lately because my university is online due to covid at the moment.)

**Reply by Navid200** [reply]
The first one does not surprise me.  But, the second one does.
You have given xDrip access to background data, right?

**Reply by scimone** [reply]
@Navid200 Where can I find this setting? Or are you talking about the data saver setting that was mentioned by mk886?

**Reply by scimone** [reply]
@Navid200 I suspected my wifi to disturb the Bluetooth so I switched it off today. No pair requests. After 3h I switched it on again. Since then another 5h passed and I still got no pair requests...

**Reply by scimone** [reply]
This seems so random. Mostly I get pair requests every other minute, but sometimes there are long breaks of silence

**Reply by Navid200** [reply]
> @Navid200 Where can I find this setting? Or are you talking about the data saver setting that was mentioned by mk886?

Android (not xDrip) Settings &#8722;> Apps & notifications &#8722;> xDrip+ &#8722;> Mobile data & Wi-Fi &#8722;> Background data

**Reply by scimone** [reply]
@Navid200 Ah right thank you. Yes it's turned on.

**Reply by Navid200** [reply]
Have you disabled Trust Auto-Connect?

**Reply by scimone** [reply]
@Navid200 Yes

**Reply by Navid200** [reply]
Did you have this problem with your previous transmitter also?

**Reply by scimone** [reply]
@Navid200 Yes same problem. Changed it 3 weeks ago

**Reply by Navid200** [reply]
How does your Garmin watch receive readings?  Is it Android wear?  Or, is it some other mechanism?

**Reply by scimone** [reply]
@Navid200 It's not android wear, the app is called Garmin Connect. I use this watchface, it gets readings from xdrip: 
Check out this app from the Garmin Connect IQ Store https://apps.garmin.com/en-US/apps/8fab3746-f56f-4b41-b3c7-5f4aaeaed6c9

**Reply by Navid200** [reply]
Do you have Nightscout?
Do you still have the older phone you had borrowed?

**Reply by scimone** [reply]
@Navid200 Yes and yes :D Why?

**Reply by Navid200** [reply]
You cannot live like this with pair requests multiple times a day.
While we wait for other suggestions, you can do something.  I am not sure it would solve the problem permanently.  I suggest:

Set up the old phone as the collector.  Set it up as a Nightscout master.
Don't reset your A21.  Don't shut it down and sore.  Rather set it up as a Nightscout follower.  Keep charging it and let it run.

Use the old phone.  
You can carry both if you like.  But, let A21 be a follower.
Does it ever stop showing readings?

**Reply by scimone** [reply]
@Navid200 That's a good idea, that's similar to what I already did those 3 days when I used the old phone as collector :)
However, I first want to explore a little bit more to see what the cause might be, also for future users with that problem.
For example I want to see if the wifi has anything to do with it. As I said before, since I turned wifi off and on again I've had a stable connections for 9h now. If I have any findings I will report them here so you can update the documentation.

**Reply by scimone** [reply]
@Navid200 Do you also have experience with the dana pump? That's the other problem that I have xD

**Reply by Navid200** [reply]
No, I don't have experience with any pumps.

**Reply by scimone** [reply]
@Navid200 Ok, I will ask in the AAPS forum. Thank you for your great support!

**Reply by Foxy7** [reply]
Maybe it's just the phone you are using, could be a glitch with it.

**Reply by markok314** [reply]
I also have the same connectivity (pairing) problems with Samsung A32. I've tried most solutions provided here but without success - pairing requests still randomly appear. I have some experience with Android programming, but not with BLE - only read a tutorial so far. I've already built xDrip on my machine, but BLE code is not trivial. Any advice about how to proceed would be helpful.

**Reply by Mayan-t** [reply]
The same phone (Samsung A32), the same problem. I've tried all the advices and settings, aldo BT tethering app. No success. Sometimes I get pairing requests every few seconds. I lived a month with, but I couldn't more. I couldn't even sleep. I had to get old Xiaomi Redmi 6A only for Dexcom.

**Comment by tolot27** [comment]
Did you try BYODA or the unmodified Dexcom app? I have no problems with a Samsung A40.

**Reply by Mayan-t** [reply]
If it a question on me:

On Samsung A32:
- original Dexcom app - installed, but the app refused to run because Samsung A32 is not a compatible phone.
- BYODA - installed, it connects to the transmitter, but it hangs up, crashes, and loses connection after a few hours. The transmitter seemed to be dead. Fortunately, it was not the case, xDrip resurrected it.
- xDrip+: frequent pairing requests problem.

But the values were very noisy (spikes, unnatural deltas, two arrows up - down - up...) compared to the smooth values from BOYDA (which I was used to with my previous phone Samsung J3 2017 or Xiaomi Redmi 6A now), but this is another problem. (But that is a pity for me because I fell in love with xDrip+)

Both the problems prevented me from using AAPS which I normally use.

I tried perhaps every possible settings combination (followed by phone restart), disabled battery optimizations for the apps, Bluetooth, etc. I also tried to change various bt developer settings.
Sorry to not have any logs, because I already uninstalled the apps from Samsung.

**Reply by tolot27** [reply]
The question was to all of you having this problem. My first guess after reading your answer is that you have a broken BT stack/chip. The BYODA app modifications does not touch the BT communication stuff. Just the device limitations are removed and some other features added. If it crashes as well then it is i more general problem not likely cannot be fixed in xDrip.

**Reply by markok314** [reply]
On my son's A32 BYODA worked, but there were still pairing  requests with approx. the same frequency as with xDrip communicating directly, so we uninstalled BYODA. The Blue Car Tethering App also didn't help. Then we tried to toggle some other settings, one by one. The last setting we changed was 'Allow OB1 unbonding' to off. Result was even worse - pairing  request appeared twice in a row. So we reverted the option and now there has been no pairing request for a week.

**Comment by tolot27** [comment]
If I understand it correctly, the problem exists at Samsung A21 and A32 and existed on A11 before but could be fixed by tweaking special access. At least BYODA works on my A40. I'll try to install xDrip but it is not a good idea to have two apps connecting to one transmitter.
@scimone Did you try BYODA? Does it crash or run fine?

**Reply by scimone** [reply]
No I haven't tried BYODA, I'm using xDrip

**Reply by scimone** [reply]
Sorry I made a mistake. I have an A22, not A21

**Comment by scimone** [comment]
Update: 
I think my problem is fixed. I haven't had any pair requests for almost 2 weeks now. When my phone is out of reach for a while, it just reconnects itself without asking.
I wish I could tell you how and why, but I didn't do anything special. I'm still using the Blue Car app (didn't dare to uninstall yet, but might try out later).
Maybe my phone had an invisible software update?

**Reply by scimone** [reply]
Uninstalled Blue Car Tethering app and still no problems

**Reply by markok314** [reply]
Now, one month and a half after my post on Feb 13 above I have a bit more experience. During this time the G6 transmitter expired, and pairing requests appeared again with the new transmitter. Of course I tried to play with setting again, but I gave up after few days. After about 20 days with the new transmitter pairing requests  stopped. No BYODA, no Blue Car App. It seems that phone needs some time 'to know' a transmitter :-(

**Comment by Navid200** [comment]
Please try the latest Nightly if you are still having connectivity problems:
https://github.com/NightscoutFoundation/xDrip/releases/tag/2022.06.09

**Reply by scimone** [reply]
Thank you.
Yes, I'm still having the issues. Sometimes I have 2 weeks of no problems at all and sometimes I have phases of getting 30 pair requests a day.
I will try the nightly and report back!

**Reply by scimone** [reply]
@Navid200 Unfortunately no improvement, I've had several pairing requests since the update

**Reply by jamorham** [reply]
@scimone can you check the xDrip event log and let me know whether you see `Android 12+ workaround` appearing in there?

**Reply by scimone** [reply]
@jamorham No there is no 'Android 12+ workaround' log. Just the 'Prompting user to notice paring request ...' and 'onReceive UPDATE ...' logs

**Reply by jamorham** [reply]
Ok this means either you are not using the latest nightly version or that the problem you're experiencing isn't the same as the one I targetted in the latest nightly which was for an issue reproduced with a pixel phone. Please double check your xDrip version via the system status page.

Assuming you are running the latest nightly, can you enable OB1 debugging by scanning this barcode using xDrip settings: https://github.com/NightscoutFoundation/xDrip/wiki/Enabling-Debugging-for-OB1-collector and then the next time you experience a re-pairing event please use the log upload feature in the xDrip event viewer and let me know that you have done so here and I will review the logs to see if there are any clues.

Please also confirm that your phone is not rooted.

**Reply by scimone** [reply]
@jamorham It says 'Version: c48ef30-2022.06.09' which seems to be the latest nightly, and my phone is not rooted.
Thank you, I have enabled OB1 debugging and will keep you updated!

**Reply by scimone** [reply]
@jamorham I just sent you the logs via the log upload tool

**Reply by jamorham** [reply]
Thanks, what these logs show is that the android framework appears to be deleting the pairing. It isn't something which xDrip is requesting and this is a different issue than seen on pixel phones.

Can you confirm my question above about whether your phone is rooted? (You would know if it was)

**Reply by scimone** [reply]
@jamorham Ok, do you have any idea what I can do about it? No my phone is not rooted

**Reply by jamorham** [reply]
I think I would need a phone of the same model to investigate the issue more deeply.

**Reply by jamorham** [reply]
Okay, looks like I will be able to obtain one of these. I will let you know how it goes.

**Reply by scimone** [reply]
@jamorham Wow thank you! Let me know if I can do anything to test

**Reply by scimone** [reply]
@jamorham btw I'm not sure anymore but I think I had similar issues with my old phone, a Samsung Galaxy s9. I will test that one again

**Reply by markok314** [reply]
@jamorham I also still have the same problems with Samsung A32 (frequent requests for pairing). I have some experience with Android programming and Linux. If I can be of any help. please let me know.

**Reply by jamorham** [reply]
@scimone how frequently do you get the pairing requests?

**Reply by scimone** [reply]
@jamorham Maybe about 10 times a day on average. But it differs a lot, some days I get 25, some days none. I've noticed it also depends on my location. In the library for example I get pairing requests every 10 minutes (maybe too many BT devices nearby? Or the wifi?)

**Reply by jamorham** [reply]
Can you also describe what other bluetooth devices you might use, do you have headset, car kit etc which might be interacting.  Would your phone be in a pocket or in a bag most of the time etc? I'm keen to simulate your environment as closely as possible.

**Reply by scimone** [reply]
@jamorham I have a garmin smartwatch but I also get the pairing requests when it's turned off. I also have a laptop and an iPad with BT on. My phone mostly lies nearby on the desk

**Reply by scimone** [reply]
@jamorham and bluetooth mouse and keyboard (when I work at home)

**Reply by jamorham** [reply]
Just to let you know I have been able to reproduce this issue on the A22. I'm going to try some experiments with this now to see if a resolution is possible.

**Reply by jamorham** [reply]
So I've dived deep in to analysis of this issue and the short answer is that this is a bug with the samsung bluetooth framework. There are about 4 different ways it can appear. I think I can work around two of them which are related to timing, but the other two I don't see how to resolve as I can't see what might cause them.

One thing I did discover during various tests is that Android 12 is available for the A22 and I wasn't able to reproduce the problem there and so if you update to android 12 then you may find the problem disappears and that they have basically fixed this.

I'm still trying out various things to see if anything can help. I would expect this bug to affect the official app as well, but because it can now go for up to 48 hours without triggering the issue, its hard for me to determine whether changes I've made have improved things and so the whole process takes a long time to prototype and test.

If anyone is interested in the technical details, the A22 uses the MTK blueooth chipset which has its own slightly customized stack (but usually pretty reliable). I believe that samsung have further customized this to store the keys in their trustzone implementation, knox etc and what basically happens is that during the bluetooth security phase when encryption is enabled and the bluetooth system process requests the key, sometimes this process fails and the key is found to be missing which causes an internal error in the bluetooth stack and it then deletes the pairing. 

This sort of thing didn't used to be a problem because we could automatically re-pair before google's security update relating to covid exposure tracking foisted manually pairing dialogs on all android versions. I've seen samsung phones losing pairing information temporarily plenty of times before. My version of AndroidAPS specifically has workarounds for this exact problem, but in those cases it didn't fail in this way where a temporary problem causes a permanent loss of the pairing information. 

I probably could patch the framework to fix this, but that would effectively require root and I don't see the point if Android 12 really is free of this problem on the A22

```
bt_btm  : Security Manager: encrypt_change status:6 State:2, encr_enable = 0
bt_btm  : btm_sec_encrypt_change: remove device, status=0x06 (KEY MISSING)
bt_btif : bta_dm_encrypt_cback :: BTM_CUSTOM_ENC_FAIL
bt_btif : bta_dm_remove_device: ACL Up count: 2
```

**Reply by markok314** [reply]
Thank you for all the effort you've put into this issue. My son's A32 automatically updated to Android 12 about a week ago, but he forgot to mention it. Based on your findings we  started xDrip on A32 again, and it has been working without pairing requests for a day now. If we can support your work in any way, please let us know.

**Reply by scimone** [reply]
Thank you for your thorough analysis @jamorham!
I'm on vacation right now and haven't had the pairing requests here so far, so they seem to be related to location or nearby devices that somehow interfere with the BT connection.
I have upgraded to Android 12 now and will keep you posted how it goes, especially when I return home.
Would be great if Google already fixed the issues themselves with the update!

**Reply by jamorham** [reply]
So the July 13th Nightly includes the code that I've been testing which I think can work-around some of the instances of this problem but it doesn't cure it completely and, because of how long it takes to test and the random nature of the problem, it is really hard to be certain how much it helps.

If you want to use this then update to the July 13th release and enable the `Special Pairing Workaround` from with the `G5/G6 Debug settings` - I have only tested this on Android 11, I'm interested to hear from people who have updated to Android 12 whether the problem is already cured for them.

**Reply by markok314** [reply]
4 days without paring request on Android 12, Samsung A32.

**Reply by jamorham** [reply]
Is this issue resolved now do we think?

**Reply by scimone** [reply]
@jamorham Hi, I haven't been at home since 4 weeks, so I unfortunately still haven't been able to test. 
I can't tell if the update which I did 10 days ago changed anything, because I haven't had any pairing requests since I left home 4 weeks ago. I'll probably return next week and will report back as soon as I can. However @markok314's experiences sound promising :)

**Reply by markok314** [reply]
Still no pairing request after 14 days. However, with the previous transmitter it happened that after about two or three weeks of pairing requests we had no problems until we changed the transmitter. Current transmitter still has 41 days to go.

**Reply by markok314** [reply]
Unfortunately some bad news. Yesterday my son updated his A32 (see versions in the screenshot) and frequent pairing requests reappeared. Then he switched his xDrip to follower for few hours and then connected to G6 again. Only two pairing requests since then. xDrip version 212205191. "Minimize scanning" is off, otherwise xDrip did not reconnect. It seems something is still not stable in Samsung's Android.
![Screenshot_20220725-174438_Settings](https://user-images.githubusercontent.com/3070254/181103969-ed346028-8ea1-4e54-9333-b8c0595ea8f6.jpg)
5

**Reply by emp-00** [reply]
@markok314 : Here on Samsung A52s - almost all the same version numbers in Software Information as in your posting. Main difference is my Kernel Version: 5.4.147-XXXX and a little bit different Build number. But absolutely the same Android 12, One UI, Play system update and Knox Versions. Since many months I have no more pairing requests. All fine! So it seems to be a Samsung mystery why it sometimes works and sometimes not. This here may indicate it's not only the Software (?)

**Reply by markok314** [reply]
@emp-00: As @jamorham indicated above, it seems to be software. It could also be that it works in some timing conditions and not others, so hardware has influence. 
@jamorham: Unfortunately we are back at frequent pairing requests today :-(

**Reply by scimone** [reply]
@jamorham Update from me: not a single pairing request since the upgrade to Android 12 on July 11th

**Reply by markok314** [reply]
@jamorham Update: Still not reliable. After we change transmitter, pairing requests appear, and then after few days they disappear. We have the latest updates for phone and xDrip from August. Anyway, repeating procedure every three months is too annoying, so I've decided to change the phone. Can you give any recommendations for a brand or specific phone in the same price range (200-300$), which works reliably with Dexcom and xDrip?
